development/compiling/introduction_to_the_buildsystem.rst

```diff
@@ -222,6 +222,18 @@ features to include/disable.
 Check the output of ``scons --help`` for details about each option for
 the version you are willing to compile.
 
+Cleaning generated files
+------------------------
+
+Sometimes, you may encounter an error due to generated files being present. You
+can remove them by using ``scons --clean <options>``, where ``<options>`` is the
```

Comment:
Is SCons config changed to remove all `.gen` files? Last time there was issue with `.gen` files, `scons --clean` was removing only `.gen` files that are valid in the current config, and issue was caused by old file that was renamed in the new version of source and was not affected by `--clean`.

Reply:
Good point, it's not going to remember generated files that are not/no longer generated by the current config.

It's still a rare occurrence that we change generated files, so IMO it should be documented as a good way to clean build objects to make a clean build. But for the gen files workaround, a full got clean is needed indeed (or `find -name "*.gen.*" -delete` which I often use).

Reply:
Well, I saw "generated" and was thinking only about `.gen.` file issues, for cleaning up object files `scons --clean` is a good option indeed. Probably both should be mentioned.

Reply:
> `find -name "*.gen.*" -delete`

At least on macOS find require path as it's first argument as it will fail with `find: illegal option -- n` error, `find . -name "*.gen.*" -delete` will work.